Шамро Павленко Михалкин 3832

kipish89 opened the pull request.

kipish89 committed:
lesson 2: add project

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,11 @@
+﻿Console.WriteLine("Работу выполнили Андрей и Роман");
+Console.WriteLine("Введите Имя");
+string n = Console.ReadLine();
```
name

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,11 @@
+Console.WriteLine("Введите Имя");
+string n = Console.ReadLine();
+Console.WriteLine("Введите Фамилию");
+string f = Console.ReadLine();
```
surname

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,11 @@
+Console.WriteLine("Введите Фамилию");
+string f = Console.ReadLine();
+Console.WriteLine("Введите дату рождения");
+int d = Convert.ToInt32(Console.ReadLine());
```
year

int.Parse

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,11 @@
+string f = Console.ReadLine();
+Console.WriteLine("Введите дату рождения");
+int d = Convert.ToInt32(Console.ReadLine());
+int nd = 2025;
```
currentYear

kipish89 committed:
command

kipish89 committed:
Merge branch 'OrlovAndrei:main' into main

kipish89 committed:
Craine repor

kipish89 committed:
Craine repos

kipish89 committed:
run01

kipish89 committed:
README.md

GambItic committed:
Update README.md

выполнение задания 3

OrlovAndrei reviewed (commented):
Исправить недочеты

kipish89 committed:
lesson 3 error

kipish89 committed:
lesson 3 er

kipish89 committed:
lesson 3 readmi vostonovlen

GambItic committed:
Update README.md

GambItic committed:
lesson 4

OrlovAndrei reviewed (commented):
подправьте форматирование
перенесите методы в конец
добавьте методы для других команд

GambItic committed:
Update README.md

kipish89 committed:
Merge branch 'OrlovAndrei:main' into main

OrlovAndrei committed:
Update .gitignore

OrlovAndrei committed:
Update .gitignore

GambItic committed:
Внесены исправления кода

Все команды перенесены в методы, методы вызываются в цикле, методы перенесены ниже цикла, исправил реадми чуть ранее

GambItic committed:
Merge branch 'OrlovAndrei:main' into main

GambItic committed:
тут

GambItic committed:
выполнил задание

GambItic committed:
Update README.md

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,427 @@
+    {
+        Console.WriteLine("Неизвестная команда. Введите 'help' для списка команд.");
+    }
+}
```
перенесите после while

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,427 @@
+}
+void ReadTask(string command)
+{
+    if (command.StartsWith("read "))
```
убрать лишнюю проверку

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,427 @@
+}
+void MarkTaskDone(string command)
+{
+    if (command.StartsWith("done "))
```
убрать

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,427 @@
+}
+void UpdateTask(string command)
+{
+    if (command.StartsWith("update "))
```
убрать

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,427 @@
+        {
+            showIndex = command.Contains("--index") || command.Contains("-i");
+            showStatus = command.Contains("--status") || command.Contains("-s");
+            showDate = command.Contains("--date") || command.Contains("-d");
```
сделать обработку комбинаций сокращенных флагов

-id -is -dis и тп

OrlovAndrei reviewed (commented):
исправить недочеты

GambItic committed:
исправил часть ошибок включающих расположение методов, переменных и некоторые другие ошибки

GambItic committed:
проверка, сделали все исправления кроме view

GambItic committed:
lesson 5 исправили все указаные ошибки

GambItic committed:
Merge branch 'OrlovAndrei:main' into main

GambItic committed:
добавили классы в отдельные файлы TodoList, TodoItem и Program

GambItic committed:
добавлен класс Profile, улучшена взаимосвязь методов и классов

GambItic committed:
lesson 6. выполнили работу, взаимосвязь всех методов и классов установлена, ошибки убраны

GambItic committed:
lesson 6

kekpit-ops committed:
Update README.md

GambItic committed:
Merge branch 'OrlovAndrei:main' into main

GambItic committed:
Merge pull request #1 from kekpit-ops/patch-1

Update README.md

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,170 @@
+Console.WriteLine($"\nДобавлен пользователь: Имя: {name}, Фамилия: {surname}, Возраст: {currentNumber}"); 
+string[] todos = new string[initialSize];
+bool[] statuses = new bool[todos.Length];
+DateTime[] dates = new DateTime[initialSize];
```
удалить

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,170 @@
+        string task = command.Substring(4).Trim(' ', '"');
+        if (!string.IsNullOrEmpty(task))
+        {
+            TodoList.Add(new TodoItem(task));
```
?

OrlovAndrei reviewed (commented):
Сделать все в одном проекте, все лишнее удалить
исправить все ошибки

GambItic committed:
поправил все, запускается)

GambItic committed:
test1

GambItic committed:
вернул флаги мы крутой

GambItic committed:
исправлены некоторые маленькие ошибки

kekpit-ops committed:
добавили icommand basecommand addcommand

kekpit-ops committed:
добавили viewcommand

kekpit-ops committed:
перенесли все методы в классы

kekpit-ops committed:
lesson 7 completed

OrlovAndrei commented on TodoList/AddCommand.cs:
```diff
@@ -0,0 +1,58 @@
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TodoList1
```
просто TodoList

OrlovAndrei commented on TodoList/BaseCommand.cs:
```diff
@@ -0,0 +1,11 @@
+
+	public TodoList todoList { get; set; }
+	public Profile Profile { get; set; }
+	public abstract void Execute();
```
убери TodoList и Profile они есть не у всех команд

OrlovAndrei commented on TodoList/CommandParser.cs:
```diff
@@ -0,0 +1,157 @@
+				return ParseDeleteCommand(inputString, todoList, profile);
+
+			case "update":
+				return ParseUpdateCommand(inputString, todoList, profile);
```
ты передаешь todoList, profile тем командам, для которых они не нужны

OrlovAndrei reviewed (commented):
Используй везде либо BaseCommand либо ICommand, определитесь и удалите лишнее
Создайте в проекте папку Commands и перенесите туда все классы команд

kipish89 committed:
ошибки исправлены

kipish89 committed:
создаю крутой класс под названием FileMAnager

kipish89 committed:
lesson 8 completed

kipish89 commented on TodoList/FileManager.cs:
5 камитов не закамитились 🦊

kekpit-ops committed:
Update README.md

kipish89 committed:
Merge branch 'OrlovAndrei:main' into main

OrlovAndrei reviewed (commented):
Перенести все классы команд в папку Commands
чтобы не было конфликтов можете поменять namespace на TodoApp

Сделать сохранение задач
исправить многострочный ввод задачи

при загрузке не нужно выводить все задачи

kipish89 committed:
trai 1

kipish89 committed:
trai 2

kipish89 committed:
trai 3

kipish89 committed:
trai 4

GambItic committed:
profileFix

GambItic committed:
taskload fix

kekpit-ops committed:
сделаны изменения

GambItic committed:
lesson9start

GambItic committed:
Statuscommit

GambItic committed:
StatusFix

GambItic committed:
StatusCommand create

GambItic committed:
statuscommand update

GambItic committed:
StatusFix

GambItic committed:
litleChanges

GambItic committed:
TableFix

kekpit-ops committed:
Update README.md

OrlovAndrei commented on TodoList/TodoList.cs:
```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+namespace TodoApp;
+public class TodoList
```
IEnumerable<TodoItem>

OrlovAndrei commented on TodoList/TodoItem.cs:
```diff
@@ -0,0 +1,113 @@
+			get => _text;
+			private set { _text = value; UpdateTimestamp(); }
+		}
+		public bool IsDone 
```
переименовать в Status

OrlovAndrei commented on TodoList/FileManager.cs:
```diff
@@ -0,0 +1,276 @@
+    		{
+        		Console.WriteLine($"[ОШИБКА] Не удалось сохранить {doneFilePath}: {ex.Message}");
+    		}
+		}
```
удалить

OrlovAndrei reviewed (commented):
исправить

kekpit-ops committed:
сделан appinfo и добавлена unexecute

kekpit-ops committed:
изменены все команды под unexecite

kekpit-ops committed:
lab 10 complete

kekpit-ops committed:
Update README.md

kekpit-ops committed:
Update README.md

OrlovAndrei commented on TodoList/Commands/Appinfo.cs:
убрать из папки Commands

назвать файл AppInfo

kekpit-ops committed:
изменения

kekpit-ops committed:
изменения

kipish89 committed:
start11

kipish89 committed:
2

kipish89 committed:
3

kipish89 committed:
lesson11 complete

kipish89 committed:
мини изменения

kipish89 committed:
еще изменея к 11

kekpit-ops committed:
Update README.md

OrlovAndrei commented on TodoList/Program.cs:
```diff
@@ -0,0 +1,155 @@
+			!(command is ViewCommand) &&
+			!(command is ReadCommand) &&
+			!(command is UndoCommand) &&
+			!(command is RedoCommand))
```
if(ShouldStoreInUndoStack(command))

OrlovAndrei reviewed (commented):
проблемы с многострочным вводом
не создается файл с задачами todos_id
нет profile --out
нет update
во view не работают комбинации флагов и флаг -a

Commit:
исправления if(ShouldStoreInUndoStack(command))

Commit:
исправления

Commit:
еще изменения

Commit:
Вывод задач

Commit:
команда для выхода из профиля

GambItic committed:
lesson12 Start

GambItic committed:
lesson12 part1 done

GambItic committed:
lesson12 2part start

GambItic committed:
lessob12 complete

kekpit-ops committed:
изменения

kekpit-ops committed:
Update README.md

kekpit-ops committed:
добавлена команда searchcommand

kekpit-ops committed:
изменена программа CommandParser

kekpit-ops committed:
lab is done

kekpit-ops committed:
1

OrlovAndrei commented on TodoList/Commands/HelpCommand.cs:
```diff
@@ -0,0 +1,41 @@
+		public override void Unexecute()
+		{
+			Console.WriteLine("Отмена команды помощи (нет изменений для отмены)");
+		}
```
можно удалить

kekpit-ops committed:
изменения

kipish89 committed:
Merge branch 'OrlovAndrei:main' into main

kipish89 committed:
исправления1

kipish89 committed:
исправления2.1

GambItic committed:
практическое задание 1/6

GambItic committed:
работа 2/6

GambItic committed:
эксептион папка работы 3/6

GambItic committed:
часть 4/6 работы, использование пользовательского исключения

GambItic committed:
исправил ошибку запуска 4.5/6

GambItic committed:
6/6 работа выполнена полностью

kekpit-ops committed:
testcases

OrlovAndrei reviewed (commented):
ok

kipish89 committed:
start lesson 3

kipish89 committed:
+ Load

kipish89 committed:
end 2.3

kipish89 committed:
Merge branch 'OrlovAndrei:main' into main

OrlovAndrei reviewed (commented):
<img width="265" height="226" alt="Image" src="https://github.com/user-attachments/assets/36cd81be-d79c-49f1-a0b8-ca2c98962adb" />

Неправильно выводит текст

kekpit-ops committed:
Idatastorage

kekpit-ops committed:
new filemanager, exeption, todoprinter

OrlovAndrei commented on TodoList/DataStorageException.cs:
в папку Exceptions

GambItic committed:
unittest part 1

GambItic committed:
test

OrlovAndrei commented:
<img width="672" height="337" alt="image" src="https://github.com/user-attachments/assets/283b1711-de93-478f-a4ac-b0da5c0910ac" />

Проект не запускается, разберитесь с ошибками в коде

GambItic committed:
test2

GambItic committed:
unit part 2

GambItic committed:
unit ComandParser inProgress

GambItic committed:
unit part 3/3 complete

GambItic committed:
start progress Server..

GambItic committed:
Server ApiData upd

GambItic committed:
Server progress..

GambItic committed:
Server done, not sync fix

GambItic committed:
try fix..

OrlovAndrei commented on TodoList.Server/Properties/AssemblyInfo.cs:
удалите

OrlovAndrei commented on TodoList.Server/App.config:
удалите

OrlovAndrei commented on TodoList.Server/TodoList.Server.csproj:
что это за бред

OrlovAndrei reviewed (commented):
<img width="323" height="528" alt="Image" src="https://github.com/user-attachments/assets/49ae30ac-10dc-4290-bac7-b3c9e9024a50" />

В коде куча ошибок

<img width="345" height="242" alt="Image" src="https://github.com/user-attachments/assets/deaa2a72-5a68-4ac9-8cbc-90f745d7d234" />

Вы очень странно сделали проект сервера, зачем там нужны assemblyinfo и app.config. Почему в .csprof так много всего и стоит древняя версия .net? Я много раз показывал как создавать проекты через visual studio, можете поискать в лекциях или загуглить как это правильно делать. Пока не исправите я не поставлю за это оценку

GambItic committed:
SERVER FIX complete 99%

kipish89 committed:
Merge branch 'OrlovAndrei:main' into main

kipish89 committed:
start тис lab 3(устоновил moq)

kipish89 committed:
iclock

kipish89 committed:
systemclock

kipish89 committed:
end тис 3

kipish89 committed:
start ркис 2.7(устоновил ef core и создал папки

kipish89 committed:
перенес в models

kipish89 committed:
создал DBContext, TodoRepository,ProfileRepository

GambItic committed:
Server, read, sync command fix

OrlovAndrei commented:
<img width="359" height="258" alt="image" src="https://github.com/user-attachments/assets/90b601f8-b730-476d-97be-4b1c30ada362" />

Тесты нерабочие

kipish89 committed:
fix test

kipish89 committed:
Merge branch 'OrlovAndrei:main' into main

kekpit-ops committed:
updated old files and csprojs

kekpit-ops committed:
new services and modules

kekpit-ops committed:
lab 9 is complete

kipish89 committed:
Merge branch 'OrlovAndrei:main' into main

OrlovAndrei commented:
Все работает, но файлы и папки у вас находятся в беспорядке, аккуратно разделите все на отдельные проекты и правильно разложите все по папкам, чтобы было легче ориентироваться

kipish89 committed:
3 работа по тестам до делана (moq)

GambItic committed:
Minimaladd API porject +

GambItic committed:
Controller add

GambItic committed:
Swagger add, mission complete
